### DISCUSSION TITLE
Will not replay when changing MacOS desktop spaces

### DISCUSSION THREAD

**Discussion by darcyvoutt** [katspaugh/wavesurfer.js#3413] [Q&A]
I'm using Wavesurfer as part of an app I'm building within the Tauri framework. I've run into an issue that I can't discern if it's something I did or is an issue with the package itself. 

To put it into context, I have two waveforms being played randomly, one muted and the other note, that uses a WebAudio/AudioContext player. This is because I am doing custom calculations to match the gain between both waveforms so someone can listen and toggle between them to hear the differences in the files.

Here's a visual for context to help make this clearer.
<img width="1364" alt="Screenshot 2023-12-15 at 9 47 33 AM" src="https://github.com/katspaugh/wavesurfer.js/assets/5162080/5c5a67ae-c46c-4b30-af0b-e94b8575a0a3">

I have a method that toggles the `playPause` of both, and everything works as expected. If I put the application out of focus and come back, it keeps working as expected. However, when I switch to another MacOS desktop space and come back, I can pause them, but the next time I try to play them again, they won't. All of my methods are working as expected. I've tested and the Wavesurfer event for playing and pausing are firing, but the actual audio and timeline cursor are doing nothing.

Looking for any insight if this could be something I'm missing or need to look into for my setup.

**Comment by katspaugh** [comment]
My 🔮 says it's something with requestAnimationFrame.

**Reply by darcyvoutt** [reply]
Not quite familiar with this. Are you implying that it's an issue with Wavesurfer or something I would need to utilize in addition to Wavesurfer?

**Reply by katspaugh** [reply]
I don't know, can't say much w/o seeing your code. My hunch is that it might be related to `requestAnimationFrame` which is known to pause when the document is out of focus.
If you can't reproduce the issue with wavesurfer on a regular web page, the problem is likely in your framework/code.

**Reply by darcyvoutt** [reply]
Thanks I'll review the documentation related to that. Just wasn't sure if that implied a bug in the package or not. The repo is private and would be hard to share pieces of the code out of context.
